sklearn/model_selection/_validation.py

```diff
@@ -455,8 +459,26 @@ def _fit_and_score(estimator, X, y, scorer, train, test, verbose,
                           for k, v in parameters.items()))
         print("[CV] %s %s" % (msg, (64 - len(msg)) * '.'))
 
-    # Adjust length of sample weights
+    # Init to empty dict if pass in as None
     fit_params = fit_params if fit_params is not None else {}
+
+    sample_weight_key, weighted_test_score = None, False
+
+    if test_score_weight is not None and test_score_weight in fit_params:
+        sample_weight_key, weighted_test_score = test_score_weight, True
+
+    if weighted_test_score:
+        train_sample_weight = _index_param_value(X,
+                                                 fit_params[sample_weight_key],
+                                                 train)
+        test_sample_weight = _index_param_value(X,
```

Comment:
Does this work?  I'm not fully convinced.  When I discovered this issue, I debugged through it and it seemed to me that `_index_param_value` was destructive to the calling parameter(s).  I had to copy the `sample_weight` in the `fit_params` in order to call `_index_param_value` two different times with `train` and `test`.